### DISCUSSION TITLE
Unit aware manipulation?

### DISCUSSION THREAD

**Discussion by durack1** [xCDAT/xcdat#335] [3. Ideas]
This newly announced [`pint` package](https://xarray.dev/blog/introducing-pint-xarray) sounds interesting. Not sure if it's more useable than `udunits`?

**Comment by pochedls** [comment]
This looks cool, @durack1. I think if there is a feature / use-case in mind, this could be migrated to: https://github.com/xCDAT/xcdat/discussions/271.

**Comment by lee1043** [comment]
I also just came across with the `pint-xarray` which I think is cool. Some example usage can be found [here](https://xarray.dev/blog/introducing-pint-xarray). I copied and pasted below use case from their website. It automatically assign proper units after variables are multiplied. It returns error when user tries addiction/subtraction of variables in different units -- which could be cool, but at the same time it could be annoying if user intended. 

```python
import xarray as xr
import cf_xarray.units
import pint_xarray

ds = xr.open_dataset('gs://cmip6/CMIP6/CMIP/NCAR/CESM2-FV2/historical/r2i1p1f1/Amon/sfcWind/gn/v20200226/', engine='zarr')
ds = ds.pint.quantify()

squared_wind = ds['sfcWind'] ** 2
squared_wind.pint.units
```

```
Out:
<Unit('meter ** 2 / second ** 2')>
```

**Reply by jypeter** [reply]
Unfortunately, this example does not mention the original units... According to the [sfcWind entry in the CMIP6 DR](https://clipc-services.ceda.ac.uk/dreq/u/4b97609a32d53dff5b8e73729e4f258b.html), I would expect: `m s-1`

The squared units should therefore be `m2 s-2` if we want a really useful result

**Comment by jypeter** [comment]
BTW, is it possible to update units of xarray objects by directly assigning a string? e.g. one of the following? That's something I regularly did with cdms2 objects
```
squared_wind.pint.units = 'm2 s-2'
squared_wind.units = 'm2 s-2'
```

It's probably possible to do so for an in-memory variable. But what about changing the units (or any other attribute) of a variable in a nc file. My apologies for this slightly off-topic question

With cdms2, I would open the file in 'r+' mode, use a file variable pointer, assign the new units and close the file.
